torch/testing/_internal/distributed/_tensor/common_dtensor.py

```diff
@@ -355,22 +355,26 @@ def backend(self) -> str:
     def build_device_mesh(self) -> DeviceMesh:
         return init_device_mesh(self.device_type, (self.world_size,))
 
-    def init_pg(self, eager_init) -> None:
-        if "nccl" in self.backend and torch.cuda.device_count() < self.world_size:
+    def init_pg(self, eager_init, backend: Optional[str] = None) -> None:
```

Comment:
Had to modify this (and below in this file) in order to allow using the "fake" backend as the root PG